### PR TITLE
fix(ci): build+push image in release flow

### PR DESCRIPTION
**Pull request by oxy**
Previously, we would build docker images on each CI run, export them, then re-import them and push them on release. However, there appears to be a bug in buildx's export flow, and images can't be re-imported into Docker.

To fix this, I've moved to building and publishing with buildx in the release workflow in one step, and removed the image build step from the main CI flow. I also moved the docker publish process to its own workflow instead of `release.yml` so that it can be re-run independently of the npm+homebrew step.

Partially adresses #3814, though a new release needs to be made for the workflow to kick in.

### CODE REVIEW

File: ci/steps/build-docker-image.sh [hunk 1]
```diff
@@ -1,12 +1,37 @@
 #!/usr/bin/env bash
```
Comment (code-asher) [inline ci/steps/build-docker-image.sh:1]: Should we rename this file? Maybe `build-and-push-docker-image.sh`?

**Review by jsjoeio — COMMENTED** [review]
As someone who is less familiar with Docker/buildx, I don't think I have enough exp to approve this.

However from a general standpoint, 
- great PR description 
- this approach feels clean and maintainable
- i appreciate you taking extra precautions and separating into it's own workflow

File: ci/steps/build-docker-image.sh [hunk 2]
```diff
@@ -1,12 +1,37 @@
...
+
+  # Download the release-packages artifact
+  download_artifact release-packages ./release-packages
+
+  # Login to Docker
+  if [[ ${CI-} ]]; then
+    echo "$DOCKER_PASSWORD" | docker login -u "$DOCKER_USERNAME" --password-stdin
+  fi
+
+  docker buildx bake -f ci/release-image/docker-bake.hcl --push
```
Comment (code-asher) [inline ci/steps/build-docker-image.sh:34]: Legend! So much better.

**Review by code-asher — APPROVED** [review]

**Comment by code-asher** [comment]
Maybe we should rename the publish workflow as well since it's not the full publish process anymore.

File: ci/steps/build-docker-image.sh (hunk 1 above)
Comment (oxy) [reply ci/steps/build-docker-image.sh:1]: I did consider this but felt it was a bit of a mouthful - though am not opposed to it!

Comment (code-asher) [reply ci/steps/build-docker-image.sh:1]: It does seem a bit long haha but I don't want to mislead anyone who might try to run this!

Comment (code-asher) [reply ci/steps/build-docker-image.sh:1]: Could maybe shorten to `docker-buildx-push.sh` or something.

Comment (oxy) [reply ci/steps/build-docker-image.sh:1]: Sounds great!

**Comment by jsjoeio** [comment]
@oxy just checking in on this. Anything we can do to help you move this forward?

**Comment by oxy** [comment]
@jsjoeio @code-asher ready for a second look and merge!

**Review by jsjoeio — APPROVED** [review]
LGTM! Great work on keeping it clean and easy to follow 👏

**Review by code-asher — APPROVED** [review]
:tada:

**Pull request merged**